### ISSUE TITLE
document behavior of signature verification - expired

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#3333]
Please help us document behavior and rendering of signatures. Please take screenshots of how the following situations are rendered:
 - [ ] a valid signature on a message
 - [ ] an invalid / bad signature
 - [ ] a signature for which we are missing the public key
 - [ ] a valid signature made with a public key when the key was still valid, but the key itself has since expired*
 - [ ] an invalid signature made with a public key when the key was still valid, but the key itself has since expired*
 - [ ] a valid signature that itself has expired, with a key that is still valid*
 - [ ] an invalid signature that has expired, with a key that is still valid*

Items where I don't remember if we have automated tests for them are marked with *, please have a look through our test suite if that particular behavior is tested and document it below.

Customer conversation for me to refer later https://mail.google.com/mail/u/0/#inbox/FMfcgxwKkbjLPppDGvKdqcjlpgshqTTr

When crafting messages / signatures / taking screenshots, please use usernames / identities that could later be published in our documentation once you take screenshots.

**Comment by bilalashraf123** [comment]
Please correct me, If I am unable to understand this requirement fully. My understanding from this requirement is, flowCypt client has a signed email and it validates the signature of the email. In order to do the signature validation, flowCrypt client performs different validation steps as mentioned above. 

We need to propose a UI for flowCrypt client about how to show the signature validation results alongwith identity infomation for the above mentioned cases. Correct?

**Comment by tomholub** [comment]
> Please correct me, If I am unable to understand this requirement fully. My understanding from this requirement is, flowCypt client has a signed email and it validates the signature of the email. In order to do the signature validation, flowCrypt client performs different validation steps as mentioned above.

The list above are not steps - they are example situations that may happen when attempting to validate a message.

> We need to propose a UI for flowCrypt client about how to show the signature validation results alongwith identity infomation for the above mentioned cases. Correct?

I would like you to only document existing behavior. I don't know exactly what the client does in all of these different situations, and I'd like you to help me document the results with screenshots.

**Comment by tomholub** [comment]
Let me know if you need any help here.

**Comment by bilalashraf123** [comment]
Do you expect something like this?

- a valid signature on a message (Will update the email content to be pasted in documentation as well. Not checked yet, how to pick the message using messageId in params)
![image](https://user-images.githubusercontent.com/24797522/106399637-bbb54800-643b-11eb-870c-9593e2415570.png)

- a signature for which we are missing the public key
![image](https://user-images.githubusercontent.com/24797522/106399661-d4256280-643b-11eb-9e39-490dddea0733.png)

No tests are available for following cases (at least not in `test/source/tests/decrypt.ts`)
- an invalid / bad signature
- a valid signature made with a public key when the key was still valid, but the key itself has since expired*
- an invalid signature made with a public key when the key was still valid, but the key itself has since expired*
- a valid signature that itself has expired, with a key that is still valid*
- an invalid signature that has expired, with a key that is still valid*

**Question:** For those scenarios, where tests are not available, should I create signed+encrypted messages using BouncyCastle/PGPainless and add these to tests as well? if yes, then which private key to use for signing and which public key for encryption?

**Comment by tomholub** [comment]
> Do you expect something like this?

Yes, except the documentation should not be using anyone's real email:

![image](https://user-images.githubusercontent.com/6306961/106446003-a4944b80-6477-11eb-9fa2-bac6211d1cbf.png)

It would be ok to edit it in the browser DOM manually (using web developer tools in the browser) just for the purposes of the screenshot from the docs. Otherwise yes - for each situation I'd like to see what are we currently doing.

> No tests are available for following cases (at least not in test/source/tests/decrypt.ts)
> Question: For those scenarios, where tests are not available, should I create signed+encrypted messages using BouncyCastle/PGPainless and add these to tests as well?

Yes - please do that so that we can test these scenarios.

> if yes, then which private key to use for signing and which public key for encryption?

You can grab any private key from the tests if it helps, or you can create a new one - either is fine, as long as it allows you to set up tests for these scenarios.

**Comment by tomholub** [comment]
There's one more scenario to document after #3366 is done - message is encrypted but not signed (which we will soon render).

**Comment by tomholub** [comment]
some notes from meeting

```ts
    ava.default(`decrypt - test signed message 1`, testWithBrowser('compatibility', async (t, browser) => {
      // 2 options
      // opt 1: manually import it (using pgp_pubkey.htm)
      const pubkeyFrameUrl = `chrome/elements/pgp_pubkey.htm?frameId=none&armoredPubkey=${encodeURIComponent(somePubkey)}&acctEmail=flowcrypt.compatibility%40gmail.com&parentTabId=0`;
      const pubkeyFrame = await browser.newPage(t, pubkeyFrameUrl);
      await pubkeyFrame.waitAndType('.input_email', 'expired.sender@example.com');
      await pubkeyFrame.waitForContent('@action-add-contact', 'IMPORT KEY');
      await pubkeyFrame.waitAndClick('@action-add-contact');
      await pubkeyFrame.waitForContent('@container-pgp-pubkey', `${recipientEmail} added`);
      // opt 2: load it into mock Public Key Server
      await BrowserRecipe.pgpBlockVerifyDecryptedContent(t, browser, {
        content: ["Message encrypted and signed using PGP/MIME."],
        params: "?account_email=flowcrypt.compatibility%40gmail.com&frame_id=frame_POsACRbHGk&message=&message_id=some-signed-msg-id&senderEmail=henry.electrum%40gmail.com&is_outgoing=___cu_false___"
      });
    }));

```

Also an idea for the future about how to make test data definition more local:

```ts
      addMessageToMockApi({
        messageId: 'some-signed-msg-id',
        content: '----BEGIN PGP SIGNED MESSAGE',
        from: ''
      })
      // definePublicKeysForThisTest({
      //   'email@example.com': '-----BEGIN ....'
      // });
```

**Comment by bilalashraf123** [comment]
With this https://github.com/FlowCrypt/flowcrypt-browser/issues/3441, which public key specifically should I use for encryption of messages?

**Comment by tomholub** [comment]
You can use this public key for encryption of messages for testing: https://flowcrypt.com/attester/pub/flowcrypt.compatibility@gmail.com

**Comment by tomholub** [comment]
It is the corresponding public key of one of the private keys in `testdata.json`.

**Comment by tomholub** [comment]
Test data from Bilal 

[PGPCrypto.zip](https://github.com/FlowCrypt/flowcrypt-browser/files/6017331/PGPCrypto.zip)
